File: discord-bot/mcp-server/src/utils/formatters.ts
/**
 * Format a date as relative time (e.g., "5m ago", "2h ago")
 */
export function formatRelativeTime(date: Date): string {
  const now = new Date();
  const diffMs = now.getTime() - date.getTime();
  const diffMins = Math.floor(diffMs / (1000 * 60));

  if (diffMins < 1) return "Just now";
  if (diffMins < 60) return `${diffMins}m ago`;
  const diffHours = Math.floor(diffMins / 60);
  if (diffHours < 24) return `${diffHours}h ago`;
  const diffDays = Math.floor(diffHours / 24);
  if (diffDays < 7) return `${diffDays}d ago`;
  return date.toLocaleDateString();
}

/**
 * Format a quantity with commas for readability
 */
export function formatQuantity(quantity: number): string {
  return quantity.toLocaleString();
}

/**
 * Format a date for display
 */
export function formatDate(date: Date): string {
  return date.toLocaleString("en-US", {
    month: "short",
    day: "numeric",
    hour: "numeric",
    minute: "2-digit",
    hour12: true,
  });
}

/**
 * Format a duration in milliseconds to human readable string
 */
export function formatDuration(ms: number): string {
  const seconds = Math.floor(ms / 1000);
  const minutes = Math.floor(seconds / 60);
  const hours = Math.floor(minutes / 60);
  const days = Math.floor(hours / 24);

  if (days > 0) {
    return `${days}d ${hours % 24}h`;
  }
  if (hours > 0) {
    return `${hours}h ${minutes % 60}m`;
  }
  if (minutes > 0) {
    return `${minutes}m`;
  }
  return `${seconds}s`;
}

/**
 * Get priority label from priority number
 */
export function getPriorityLabel(priority: number): string {
  switch (priority) {
    case 0: return "Low";
    case 1: return "Medium";
    case 2: return "High";
    case 3: return "Critical";
    default: return "Unknown";
  }
}

/**
 * Format stockpile type for display
 */
export function formatStockpileType(type: string): string {
  switch (type) {
    case "SEAPORT": return "Seaport";
    case "STORAGE_DEPOT": return "Storage Depot";
    case "DEPOT": return "Depot";
    case "BASE": return "Base";
    default: return type;
  }
}

File: mcp-server/src/foxhole/formatters.py
"""Formatting utilities for display."""

from datetime import datetime


def format_relative_time(date: datetime) -> str:
    """Format a date as relative time (e.g., '5m ago', '2h ago')."""
    now = datetime.utcnow()
    diff = now - date
    diff_mins = int(diff.total_seconds() / 60)

    if diff_mins < 1:
        return "Just now"
    if diff_mins < 60:
        return f"{diff_mins}m ago"
    diff_hours = diff_mins // 60
    if diff_hours < 24:
        return f"{diff_hours}h ago"
    diff_days = diff_hours // 24
    if diff_days < 7:
        return f"{diff_days}d ago"
    return date.strftime("%b %d, %Y")


def format_quantity(quantity: int) -> str:
    """Format a quantity with commas for readability."""
    return f"{quantity:,}"


def format_date(date: datetime) -> str:
    """Format a date for display."""
    return date.strftime("%b %d, %I:%M %p")


def format_duration(ms: int) -> str:
    """Format a duration in milliseconds to human readable string."""
    seconds = ms // 1000
    minutes = seconds // 60
    hours = minutes // 60
    days = hours // 24

    if days > 0:
        return f"{days}d {hours % 24}h"
    if hours > 0:
        return f"{hours}h {minutes % 60}m"
    if minutes > 0:
        return f"{minutes}m"
    return f"{seconds}s"


def get_priority_label(priority: int) -> str:
    """Get priority label from priority number."""
    labels = {
        0: "Low",
        1: "Medium",
        2: "High",
        3: "Critical",
    }
    return labels.get(priority, "Unknown")


def format_stockpile_type(stockpile_type: str) -> str:
    """Format stockpile type for display."""
    type_labels = {
        "SEAPORT": "Seaport",
        "STORAGE_DEPOT": "Storage Depot",
        "DEPOT": "Depot",
        "BASE": "Base",
    }
    return type_labels.get(stockpile_type, stockpile_type)
